---
title: "Individual homework BOHTA 2019"
output:
  html_document:
    df_print: kable
  pdf_document: default
---

__Name:__ Nuttapong Mekvipad

__KI id:__ tpx555

__Exam ID:__ 

#Instructions

For deadlines etc, see Digital Exam

You have to supply both the answer (numbers, tables, plots, discussion or combinations thereof) as well as the R or Unix code you used to make the plots/results. This should be done using this R markdown template: we want both the R markdown file and the resulting __html__ file (__not a PDF!__). The html file is what you will present at the oral exam - it will be loaded on a computer/beamer when you arrive. At the oral exam, there may be questions on the homework and/or anything else from the course.

In the home work part, all aids are allowed except working with others since it is an individual exam. In the oral exam part, no aids are allowed except your html file.  

Please submit the results to both digital exam and absalon, to be safe. 

Note that:

1. If the R code or UNIX commands code gives different results than your results, you will get severe point reductions or even 0 points for the exercise

2. Some questions may request you to use R or UNIX commands /options we have not covered explicitly in the course: this is part of the challenge

3. Much like in an academic paper, the analysis and results should be presented on a level of detail that someone else could replicate the analysis. If not, we will deduct points.

4. Use tidyverse for all analyses and plotting, unless we explicitly used something else in class for the relevant plot or analysis (e.g. plotting trees, expression analysis on matrices, heat maps, etc)

For statistical tests, you have to:

1. Motivate the choice of test 

2. State exactly what the null and alternative hypothesis are (depends on test!) and the P-value threshold

3. Comment the outcome: do you reject the null hypothesis or not, and what does this mean for the actual question we wanted to answer (interpretation)?

## Questions 0:

__Question 3.0.1: Load ALL the packages you need for the entire assignment, and set your seed to 2018 using `set.seed(2018)`. Tip: use `{r message=FALSE, warning=FALSE}` in the  R markdown code to avoid spamming the report with non-helpful messages__

```{r message=FALSE, warning=FALSE}
library(tidyverse)
library(ggrepel)
library(DESeq2)
set.seed(2019)
theme_set(theme_bw())
```

## Question 1:

To get bed files containing RefSeq track and CpG island track of hg18, we used "Table browser" menu in Genome browser to download both tracks of assembly hg18.

The -300 to -1 regions around TSSs were selected using below command. The option -l was for specifying the size of region upstream (300 bp upstream here), -r was for region downstream and option -s was for specifying bedtools to select region based on strand.

```{bash eval=FALSE}
bedtools flank -i hg18_refseq.bed -g human.hg18.genome.txt -l 300 -r 0 -s > hg18_refseq_ups300.bed
```

__Question 1.1.  What percentage of promoter regions overlap CpG islands with at least 1 bp?__

To find the percentage of promoter regions that overlap CpG islands with at least 1 bp overlapping, we used bedtools intersect command as in below chunk. The hg18_refseq_ups300.bed which contained promoter region was intersected with hg18_cpg.bed. The option -v in command told bedtools to return promoters without any overlap with CpG island, and the option -wa in command told bedtools to return original promoters with overlap. Then the number of unique lines of intersection results was counted.

In total 57.54 percent of promoter regions (41838 from 72711 promoters) overlapped CpG islands which was close to our expectation at roughly around 50 percent. 

```{bash eval=FALSE}
bedtools intersect -v -a hg18_refseq_ups300.bed -b hg18_cpg.bed | sort | uniq | tee PromNoOverlap.bed | wc -l # 30873
bedtools intersect -wa -a hg18_refseq_ups300.bed -b hg18_cpg.bed | sort | uniq | tee PromOverlap.bed | wc -l # 41838
wc -l hg18_refseq_ups300.bed # 72711
# sum up to total number
# overlapping promoter percentage = 41838/72711 * 100 = 57.54%
```


__Question 1.2. What percentage of CpG islands overlap promoter regions with at least 1 bp?__

Similar to question 1.1, bedtools intersect was also used here, but the input to -a option was changed to file containing CpG island and input to -b option to file containing promoter regions.

In total 46.71 percent of CpG isalnd regions (13183 from 28226 CpG islands) overlapped promoter regions. The result was also close to our expectation at roughly 50 percent. 

```{bash eval=FALSE}
bedtools intersect -v -a hg18_cpg.bed -b hg18_refseq_ups300.bed | sort | uniq | tee CpGNoOverlap.bed | wc -l # 15043
bedtools intersect -wa -a hg18_cpg.bed -b hg18_refseq_ups300.bed | sort | uniq | tee CpGOverlap.bed | wc -l # 13183
wc -l hg18_cpg.bed # 28226
# sum up to total number
# overlapping CpG percentage = 13183/28226 * 100 = 46.71%
```

__Question 1.3. Do refseq genes (defined as rows in the refseq track) whose promoter region overlap CpG islands have significantly more or less exons than the ones whose promoters  do not overlap CpG islands? What is the difference in the exon count d distributions between these two gene sets?__

```{r message=FALSE, warning=FALSE}
overlapPromoter <- read_tsv("PromOverlap.bed", col_names = FALSE) # number of exon at col 10
notoverlapPromoter <- read_tsv("PromNoOverlap.bed", col_names = FALSE)
```


Before we chose what kind of test to use, we needed to see the distribution of the number of exon of genes with and without overlap between CpG island and promoter. We checked this by plotting histogram below.

```{r}
exonCountOverlap <- overlapPromoter[,10] %>% add_column(type = "Overlap")
exonCountNoOverlap <- notoverlapPromoter[,10] %>% add_column(type = "No Overlap")

bind_rows(exonCountOverlap, exonCountNoOverlap) %>%
  ggplot(aes(x=X10)) + geom_histogram(bins=40) + facet_wrap(~type, scales = "free") +
  xlab("exon count")
```

We can see that the distribution shape of number of exon in both groups were not normal shape, so we could not use t-test to test the difference in number of exon here. We could use either Chi-square test or Fisher exact test for hypothesis testing.

The particular hypothesis that we wanted to test was that genes whose promoters do not overlap CpG islands are more prone to only have one exon. To test this hypothesis, we tested whether the proportion of genes with more than one exon is larger in genes with overlap between CpG island and promoter than that proportion in gene without overlap, or not. We also had to confirm that these proportions were significantly different using Chi-square or Fisher's exact test.

To see the proportion, we created contigency table as below.

```{r message=FALSE, warning=FALSE}
countOverlap <- overlapPromoter[,10] %>% 
  group_by(single.exon = X10 < 2) %>% summarise(count.overlap=n())
countNoOverlap <- notoverlapPromoter[,10] %>% 
  group_by(single.exon = X10 < 2) %>%
  summarise(count.not.overlap=n())

contig_table <- full_join(countNoOverlap, countOverlap) 

# convert to matrix to transpose
contig_mat <- t(as.matrix(contig_table[,2:3]))
colnames(contig_mat) <- c("not.single.exon", "single.exon")
contig_table <- as.tibble(contig_mat, rownames = NA) %>% 
  mutate(proportion.single = single.exon/rowSums(.))

# fischer test need matrix
temp_mat <- as.matrix(contig_table)
row.names(temp_mat) <- row.names(contig_mat)
contig_mat <- temp_mat
contig_mat
```

We can see that the proportion of single exon genes in the group of genes without overlap between CpG island and promoter region was about 6 times larger than the group of genes with overlap. However, we needed to test whether this difference in proportion was statistically significant at the certain alpha level (here we used 0.05). 

To test difference between two proportion, we used Fischer's exact test with null hypothesis that the proportion of single exon genes in the group of genes with overlap between CpG island and promoter region was not different from that proportion of the group of genes without overlap between those two regions. The alternative hypothesis is that the proportion of single exon genes in the group of genes with overlap between promoter and CpG island was significnatly different from that proportion of the group of genes without overlap.

```{r}
fisher.test(contig_mat[,1:2]) 
```

From the results, we can see that the p-value was 2.2e-16 which was much lower than the threshold at 0.05. Therefore, we rejected null hypothesis which means that the proportion of single exon genes in the group of genes with overlap between CpG island and promoter region was significantly different from that proportion of the group of genes without overlap. 

Therefore, from the proportion size difference in contigency table and Fisher's exact test result, we can say that the propotion of single exon genes is significantly higher by around 6-fold in the group of genes without overlap than the genes with overlap.

__Question 1.4. Can you identify the most important problem with the analysis above (so, qustions 1.1 to 1.3) - and how would you solve it (only discussion text needed,   maximum of 50 words)?__

The most important problems here was that we counted 2 sites to be overlap even when there was just 1 bp or more overlap. This might result in overestimation of the proportion of promoters that had meaningful biological association with CpG island. To solve this problem, we can change overlaping criteria by specifying minimum length of overlap to be some fraction of promoters length using -f option. 


## Question 2

__Question 2.1: Using tidyverse only (no matrix operations, aggregate, etc),  make a SINGLE plot ( using `ggplot` and `facet_wrap`) showing the average CAGE signal  on the top panel and the average p300 signal  on the bottom panel, also showing the with standard error of the mean in both panels. Specifically: __

__Y = average CAGE or P300 ChIP signal per bp. __

__X = positions in bp  (+-1000 from enhancer center). __

__Make sure to distinguish the signals by strand for CAGE data. __

__Because we plot means, we would also like to have a measure of the variance - in this case the standard error of the mean. Unfortunately there is no R function for this, so you will have to make a custom function for this.  The formula for standard deviation of a vector called `x` is `sd(x)/sqrt(length(x))`. Add a ribbon (`geom_ribbon()`) to the plot which shows the +- standard error of the mean for each calculated mean value.__

To create the plot, we first appended data.source column to CAGE and p300 tibble to specify source of data using dplyr's add_column(). We also appended strand column with value set to "Unspecified" to p300 tibble. We then used dplyr's bind_rows() to concatenate p300 and cage tibble into 1 tibble for plotting. The concatenated tibble was change into long format using gather(). We grouped the data by position, data.source and strand, and then we calculated mean, and mean $\pm$ standard error of each group.


__Comment on your plot. What are we actually seeing? Why is this important (150 words max)?__

From the both CAGE and ChIP plot, we can see that mean intensity of signal increased as the positions became closer to position 0, but the peak positions of both plots were different. For ChIP, the peak intensity was at position 0, while for the CAGE the peak intensity was at around 100 bp downstream of position 0 (around -100 on minus strand, and at around +100 on plus strand). We can also see that the variance in intesity of TSS signal was larger than variance of p300 ChIP signal.

What we see here is important as it actually told us about relative position of TSS and the center of enhancer region. From the results, TSS was mostly located approximately 100 bp downstream of enhancer region on both strands, but TSS could also be found around 100 bp upstream and 400 bp downstream from center of enhancer. The results also told us that the position of p300 binding site in enhancer region was quite varied, but it was mostly found at center of enhancer region. These results also agreed with and confirmed our knowledge about typical gene structure that TSS is located downstream of enhancer. In addition, we could see that the distance between TSS and enhancer center can vary between genes. 

```{r message=FALSE, warning=FALSE}
cage_dat <- read_tsv("cage.txt")
p300_dat <- read_tsv("p300.txt")

cage_dat <- add_column(cage_dat, data.source = "CAGE")
p300_dat <- add_column(p300_dat, data.source = "ChIP", strand = "Unspecified")

plotDF <- bind_rows(cage_dat, p300_dat) %>% 
  gather(key = "position", value = "TPM", -data.source, -strand, -id) %>%
  group_by(position, data.source, strand) %>% 
  summarise(mean = mean(TPM),  
            min.ribbon = mean - sd(TPM)/sqrt(n()),
            max.ribbon = mean + sd(TPM)/sqrt(n())) %>% ungroup() %>%
  mutate_at("position", as.numeric)

ggplot(plotDF) + 
  geom_ribbon(aes(x=position, ymin = min.ribbon, ymax = max.ribbon, group=strand), 
              fill = "grey80") +
  geom_line(aes(x=position, y=mean, group=strand, color=strand)) +
  facet_wrap(~data.source, nrow = 2, scales = "free") + 
  scale_color_manual(values=c("red", "blue", "yellow")) + ylab("Average signal")
```
** To see the ribbon of ChIP plot, we need to zoom in closely as the mean $\pm$ standard error range is very narrow here.

## Question 3

#### Part 1: Exploratory Data Analysis

__Question 3.1.1: Read all dataset into R, and make sure all the study design, count matrix and normalized matrix have matching numbers / names of both samples/treatments and genes.__

We read studyDesign.tsv and diabetesGenes.tsv file using read_tsv(), but we had to read countMatrix.tsv and normalizedMatrix.tsv using read.table, as read_tsv could not handle file which the number of column name in header are not equal to number of column (e.g. the file with row names). Then we printed out head rows of each tibble or data frame to check the integrity. We can see that all files were read in correctly, and the number and name of samples in design tibble were the same as in other data frame.


```{r message=FALSE, warning=FALSE}
studyDesign <- read_tsv("studyDesign.tsv")
# read_tsv cannot handle file with first col as row names
countMat <- read.table("countMatrix.tsv", row.names=1, header = TRUE) 
normCount <- read.table("normalizedMatrix.tsv", row.names=1, header = TRUE)
diabetesGene <- read_tsv("diabetesGenes.tsv")

studyDesign
head(countMat)
head(normCount)
head(diabetesGene)

# check number of genes
dim(countMat)
dim(normCount)
```

__Question 3.1.2: Using the normalized matrix, perform a hierachical clustering on the samples using euclidian distance and average linkage, and plot the resulting tree.__

To perform a hierachical clustering, we calculated distance matrix using dist(). The method for distance calculation was set to euclidean distance (method = "euclidean"). Then, we performed hierachical clustering using distance matrix. The method for clustering was set to average linkage (method = "average"), and the hierachical clustering result was plotted below.

From plot below, we can seen that there were 2 main clusters which were the cluster of samples from treatment group (sample6-10) and the cluster of samples from control group (sample1-5). We can also see that the samples from treatment group had small distance (short tree length) between each other, while sample 5 in the control group was seperated by a large distance from other 4 samples in the control group.  


```{r}
distMat <- dist(t(normCount), method = "euclidean")
hclustRes <- hclust(distMat, method = "average")
plot(hclustRes, xlab = "hierachical clustering results", sub="")
```

__Question 3.1.3: Using the normalized matrix, perform a Principal Components Analysis (PCA) on the samples and produce a PCA-plot of the two first components, where the axis labels show the amount of variance explained by each component and samples are colored by their experimental group. Find a way to label the samples, so the identity (the sample name) of each point can easily be seen (hint: look at `geom_text()` or the ggrepel package!). Note, you should center but not scale the data.__

We then performed PCA on the samples using default parameters (center = TRUE, sacle. = FALSE) and plotted the first and second principal components using ggplot. The labels of the samples' condition were appended to PCA projection results using dplyr's bind_cols() before we plotted using ggplot. The x axis was PC1 and y axis was PC2. The points were color by condition (treatment or control), and the point labels were also added to the plot using geom_text_repel function from ggrepel with some modification in repel parameters to reduce line thickness and increase distance between labels (segment.size = 0.25, box.padding = 0.6). 

From the results, we can see that all samples from treatment groups formed single cluster, while samples from control group formed 2 separate clusters which were the cluster containing samples 1 to 4 and the cluster containing only sample 5.  


```{r}
pca_sample <- prcomp(t(normCount)) 
as_tibble(pca_sample$x) %>% bind_cols(studyDesign) %>%
  ggplot(aes(x=PC1, y=PC2, color = Condition, label = rownames(pca_sample$x))) +
  geom_point() + 
  geom_text_repel(segment.size = 0.25, box.padding = 0.6)
```

__Question 3.1.4: Based on the two previous questions, discuss (max 70 words) whether your observations indicate the presence of one or more failed sample preparations. Use the supplied count matrix to explore whether you believe the sample should be discarded prior to further analysis.__

Based on PCA and hierachical clustering results, we can see that sample 5 which belonged to control group was not clustered with other samples from control group. Even hclust put sample 5 in the same cluster as other control samples, the distance of sample 5 was far from other 4 samples. When we looked at sum of the read count of each sample, we can see that the number of read in samples 5 was over 1 order of magnitude lower than other 4 control samples (and also lower than treatment group). Therefore, we believed that sample 5 was failed sample and it should be discarded. 

```{r}
as.data.frame(colSums(countMat))
```

#### Part 2: Differential Expression (DE)

__Question 3.2.1: Discard any outlier(s) identified in Part 1 and use DESeq2 to obtain differentially expressed (DE) genes between the two experimental conditions. Use default parameter, except use a logFC threshold of 0.25 and an adjusted P-value threshold of 0.05.__

To obtain differentially expressed genes usind DESeq2, we converted expression matrix data to DESeq-usable data structure using DESeqDataSetFromMatrix function. The matrix that we supplied to countData argument of DESeqDataSetFromMatrix() must be matrix of unnormalized non-negative integer values of counts as specifed in DESeq2 vignette. So, we supplied the matrix from countMatrix.tsv to the function. The column 5 corresponding to sample 5 was removed from count matrix before it was input to function. The row corresponding to sample 5 was also removed from design matrx before supplied it to colData argument.      

```{r message=FALSE, warning=FALSE}
ddsnormCountNoOut <- DESeqDataSetFromMatrix(countData = countMat[,-5],
                                            colData = studyDesign[-5,],
                                            design = ~ Condition)

ddsnormCountNoOut
```

We next called DESeq function over the data read in by DESeqDataSetFromMatrix. All of the parameters for DE analysis were kept as default values. Then we extracted results from DE analysis using results(). The comparision were set to compare between treatment and control group. The log2 fold change cutoff was set to 0.25, and the cutoff for adjusted p-value was set to 0.05. Other parameters for results extraction were kept as default.

```{r message=FALSE, warning=FALSE}
ddsnormCountNoOut <- DESeq(ddsnormCountNoOut)
ctrlVsTrt <- results(ddsnormCountNoOut,
                     contrast = c("Condition", "Trt", "Ctrl"),
                     lfcThreshold = 0.25,
                     alpha = 0.05)
```



__Question 3.2.2: Convert the output of DESeq2 to a tibble, and make an MA-plot using ggplot2. The MA-plot should show the overall trend using a trend line and genes should colored according to their DE status. __

We converted the extracted DE results and used it to plot the MA-plot. The x axis of MA-plot was base mean of each transcript, and the y axis was log2 fold change between conditions. We colored the points by adjusted p-value (red for padj >= 0.05, blue for padj < 0.05 and grey for the transcript that padj could not be calculated). The expected trend line was plotted in red and the fitted trend line was plotted in blue using geom_smooth function.

__Discuss whether the MA-plot indicates an approriate DESeq2 analysis (max 70 words discussion).__

From the result, we can see that the fitted trend line curved upward and did not follow the expected trend line. We can also see that the distribution of point on MA-plot was not symmetric. The highly expressed genes were upregulated more than expected. This indicated the problem in normalization step during DESeq analysis. We can also see some data point that adjusted p-value cannot be caluculated due to problem with read count e.g. zero counts or extreme count outlier [1].

```{r message=FALSE, warning=FALSE}
dex_tibble <- ctrlVsTrt %>% as.data.frame %>%
  rownames_to_column(var = "Gene") %>% as_tibble() 

dex_tibble %>% ggplot(aes(x=baseMean, y=log2FoldChange)) + 
  geom_point(aes(color= padj<0.05),alpha = 0.2) +
  geom_smooth() + scale_x_log10() + 
  geom_hline(yintercept = 0, color = "red")

```



__Question 3.2.3: Report the number of upregulated genes and the number of downregulated genes (regardless of whether the MA plot above looks good or not).__

The number of significantly upregulated gene in treatment could be found using code in below chunk. We filtered for the upregulated gene using the same cutoff value as above analysis. Genes would be count as siginificantly upregulated or down regulated when log2 fold change were higher than 0.25 or lower than -0.25 respectively, and the  adjusted p-value must be lower than 0.05. In total, there were 1172 upregulated genes and 1165 downregulated genes.


```{r}
n_upreg <- dex_tibble %>% filter(padj < 0.05, log2FoldChange > 0.25) %>% 
  nrow() # n upreg = 1172
n_downreg <- dex_tibble %>% filter(padj < 0.05, log2FoldChange < -0.25) %>% 
  nrow() # n downreg = 1165

as_tibble(list(status=c("upregulated", "downregulated"), n=c(n_upreg,n_downreg)))
```

__Question 3.2.4: Sort the DE statistics table  that you get from DESeq2 to report the top 5 genes sorted by__ 

__a) positive logFC (highest on top)__ 

__b) negative logFC (lowest on top)__

__only looking at significantly differentially expressed genes__


__a) positive logFC (highest on top)__ 

To get top 5 significantly differentially expressed genes with highest log2 fold change, we filtered DE analysis tibble extracted by results() for genes with adjusted p-value lower than our threshold at 0.05. Then we sorted tibble by log2 fold change values from highest to lowest. The top 5 significantly upregulated gene were Gene7019, Gene8748, Gene8590, Gene136 and Gene2627, respectively.

```{r}
dex_tibble %>% filter(padj < 0.05) %>% arrange(desc(log2FoldChange)) %>% head(5)
```

__b) negative logFC (lowest on top)__

To get top 5 significantly differentially expressed genes with lowest log2 fold change, we applied similar method as 3.2.4.a, but instead we sorted tibble by log2 fold change values from lowest to highest. The top 5 significantly downregulated gene were Gene9361, Gene1826, Gene2320, Gene8196 and Gene989, respectively.

```{r}
dex_tibble %>% filter(padj < 0.05) %>% arrange(log2FoldChange) %>% head(5)
```

#### Part 3: Gene Set enrichment 

__Question 3.3.1: Novo Nordisk claims their treatment affects expression of genes related to diabetes. Your task is to investigate whether this is true. They have supplied you with a long list of genes that are diabetes-related - diabetesGenes.tsv. One way to think about this file is that it lists all genes with a "diabetes" GO-term. Make a 2x2 contingency table that shows:__

```{r echo=F, message=FALSE, warning=FALSE}
table(`DE gene`=c(T, F, F, T), `Diabetes gene`=c(T, F, T, F))
```
__where the 1's are replaced by the actual counts. Hint: there are many ways of matching two vectors. One way is based on  `vector1 %in% vector2`  (we used it in the tidyverse lectures).__

To make 2x2 contigency table showing information above, we first created boolean vector of diferentially express status (true = significantly DE, false = not significantly DE) using the criteria for DE genes as above (padj < 0.05 and log2 fold change either > 0.25 or < -0.25). Then, we created boolean vector of status of being diabetes gene (true = being diabetes gene, false = not being diabetes gene) using %in%. Finally we put both vector in template above to obtain table as above. 

In total, 8762 genes were counted in this table excluding 762 genes that adjusted p-value could not be calculated (padj = NA). There were 122 diabetes genes, 89 of them were not significantly diferentially expressed and 33 of them were significantly diferentially expressed. There were 8640 non-diabetes genes, 6336 of them were not significantly diferentially express and 2304 of them were significantly diferentially express.


```{r message=FALSE, warning=FALSE}
dex_tibble_woNA <- dex_tibble %>% filter(!is.na(padj))
DEstat <- dex_tibble_woNA$padj < 0.05 & (dex_tibble_woNA$log2FoldChange > 0.25 | dex_tibble_woNA$log2FoldChange < -0.25)
isDiabetes <- dex_tibble_woNA$Gene %in% diabetesGene$Gene

DExDia <- table(`DE gene`=DEstat, `Diabetes gene`=isDiabetes) 
# sum up to 8762 (all genes with padj not NA)
DExDia
```

__Question 3.3.2: Once you have your 2x2 table, investigate whether differentally expressed genes are significantly _overrepresented_  (or, in other words,  _enriched_) in terms of diabetes genes. Report both the 2x2 contingency table used for the test and P-value, and discuss the result ( max 150 word discussion).__ 

__Analysis descritption__

Contigency table obtained from question 3.3.2 was converted to 2x2 matrix with row and column name, so that we can easily convert it further to 2x2 tibble. We than calculated the proportion of diabetes genes in groups of differentially expressed and non-differentially expressed genes to see effective size. The resulting tibble was converted back to matrix so that is can be input to fisher.test() function. The contigency table for Fisher, exact test was as below.

We used one-sided Fisher's exact test to test the different in proportion with alpha = 0.05. We chose one-side test, beacuse here we only interested in testing whether the proportion of diabetes genes in differentially expressed genes was larger (was enriched) than in non-differentially expressed genes or not. So that if the proportion is significantly larger, we can say that diabetes genes were overepresented in differentially expressed genes. Also, we did not interest in the case where that proportion in non-differentially expressed genes was larger than in differentially expressed genes as we wanted to see whether diabetes genes is enriched or not.

The null hypothesis was that the proportion of diabetes genes in the group of diferentially expressed genes was not different from the proportion of diabetes genes in the group of non diferentially expressed genes.

The alternative hypothesis was that the proportion of diabetes genes in the group of diferentially expressed genes was larger than the proportion of diabetes genes in the group of non diferentially expressed genes.

```{r}
# convert R table to 2x2 matrix so that data can be easily handle (convert to tibble)
contigDEdf <- as.data.frame.matrix(DExDia) 
rownames(contigDEdf) <- c("not.DE.gene","DE.gene")
colnames(contigDEdf) <- c("not.diabetes", "diabetes")


# calculate proportion of DE 
contigDEdf <- contigDEdf %>% as.data.frame() %>% 
  rownames_to_column() %>% as_tibble() %>% 
  mutate(proportion.DE = diabetes/rowSums(.[,2:3])) 

# convert to matrix for Fisher's exact test
contigDEmat <- as.matrix(contigDEdf[,2:4]) 
rownames(contigDEmat) <- contigDEdf$rowname

# change order of row and column to easily understand fisher results
contigDEmat <- contigDEmat[nrow(contigDEmat):1,]
contigDEmat <- contigDEmat[,c(2,1,3)]
contigDEmat
```

__Discussion (max 150 word discussion)__

From the contigency table above, we can see that the proportion of diabetes genes in group of diferentially expressed genes is greater than the proportion of diabetes genes in the group of non diferentially expressed genes. However, the diffrence was very small (very small effective size). 

The one-side Fisher's exact test results confirmed that this difference was not significant. We can see that the p-value from the test was 0.497 which was much larger than our alpha = 0.05. So, we retained null hypothesis that the proportion of diabetes genes in the group of diferentially expressed genes was not different from the proportion of diabetes genes in the group of non diferentially expressed genes. Therefore, the Novo Nordisk's claims that the treatment affects expression of diabetes-related genes is not true here. 


```{r}
fisher.test(contigDEmat[,1:2], alternative = "greater")
```

### References

1. Love, M.I., Anders, S., and Huber, W. (2019, February 9). *Analyzing RNA-seq data with DESeq2.* Retrieved from https://bioconductor.org/packages/release/bioc/vignettes/DESeq2/inst/doc/DESeq2.html#pvaluesNA



